frontend/src/contexts/AuthContext.tsx

```diff
@@ -35,7 +41,15 @@ export const useAuth = (): AuthContextProps => {
 
 // Provider hook that creates auth object and handles state
 const useProvideAuth = () => {
-  const [isAuthenticated] = useLocalStorage<boolean>(LOGGED_IN_KEY)
+  // TODO #4279: Remove after React rollout is complete
+  const { data: user } = useQuery<UserDto>(
```

Comment:
Comment: could we use the useUser hook directly? Or does the use user hook look at the localStorage value before enabling? Maybe we can add an override enabled prop if that is the case

Reply:
>  Or does the use user hook look at the localStorage value before enabling? 

Yup correct

> Maybe we can add an override enabled prop if that is the case

ok i try

Reply:
> Or does the use user hook look at the localStorage value before enabling? 

the `useUser` hook uses `useAuth` to get the localStorage value (so there's a circular reference). To get around this I replaced `useAuth` with `useLocalStorage` in `useUser`

> Maybe we can add an override enabled prop if that is the case

Ok! added this too

Reply:
ah, now it looks messy. i reviewed it without completely looking at the hook. suggest reverting haha ><. The previous code was more readable, and considering this is "throwaway" code i think we can get away with some duplication.

or we can set the state of isAuthenticated to be contingent on return value of useUser instead of the other way round:

```tsx
export const useUser = (): UseUserReturn => {
  const [isAuthenticated, setIsAuthenticated] = useLocalStorage<boolean>(LOGGED_IN_KEY)

  const { data: user, isLoading } = useQuery<UserDto>(
    userKeys.base,
    () => fetchUser(),
    // 10 minutes staletime, do not need to retrieve so often.
    { 
      staleTime: 600000,
      // Removed enabled state, just retrieve every 10 minutes
      // new functions, update auth states
      onSuccess: () => setIsAuthenticated(true),
      onError: () => setIsAuthenticated(undefined) // remove login on error, may not be needed since we already have ApiService
    },
  )

  return {
    user: isAuthenticated ? user : undefined,
    isLoading,
  }
}

```

Reply:
haha it's ok! I will revert :) I think that's the least messy / easiest to remove after the rollout